Documentation/Development.md

```diff
@@ -6,27 +6,232 @@
+implementation.
+
+There are prebuilt versions of the X10 library for certain platforms. If a
+prebuilt is unavailable for your desired platform, you can build X10 from
+source following the instructions below.
+
+- [Windows 10 (x64)](https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC80Mzc2OC9hcnRpZmFjdE5hbWUvdGVuc29yZmxvdy13aW5kb3dzLXg2NA2/content?format=zip)
+- [macOS (x64)](https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC80Mzc2OC9hcnRpZmFjdE5hbWUvdGVuc29yZmxvdy1kYXJ3aW4teDY00/content?format=zip)
+
+The location where you extract the prebuilt library is important to remember
+as it will be required in building the TensorFlow Swift APIs.
+
+#### (Option 2) Building X10
+
+#### Requirements
+
 * Bazel. This can be installed [manually][bazel] or with
 [Bazelisk][bazelisk]. You will need a version supported by TensorFlow
 (between `_TF_MIN_BAZEL_VERSION` and `_TF_MAX_BAZEL_VERSION` as specified in
 [tensorflow/configure.py][configure.py]).
 * Python3 with [numpy][numpy].
-* CMake.  CMake 3.16 or newer is required to build with CMake.
 
-### Building and testing
+#### Building
+
+The X10 implementation is distributed as part of the Swift for TensorFlow APIs
+repository. It consists of two halves:
+
+1. [XLA Client](Sources/x10/xla_client)
+1. [X10](Sources/x10/xla_tensor)
+
+The library is designed to be built as part of the
+[tensorflow](https://github.com/tensorflow/tensorflow) build. As such, in
+order to build X10, you must build tensorflow.
+
+Currently X10 is developed against TensorFlow 2.4.0. The following build
+scripts provide commands to build on common platforms. They largely replicate
+the build instructions for TensorFlow. The instructions diverge in that we
+must copy the additional X10 library sources into the tensorflow repository.
+The following table identifies the copied locations:
+
+| swift-apis source | tensorflow destination |
+|-------------------|------------------------|
+| Sources/CX10      | swift_bindings         |
+| Sources/x10/xla_client | tensorflow/compiler/xla/xla_client |
+| Sources/x10/xla_tensor | tensorflow/compiler/tf2xla/xla_tensor |
+
+We build two specific targets:
+1. `//tensorflow:tensorflow`
+2. `//tensorflow/compiler/tf2xla/xla_tensor:x10`
+
+On Windows, we build the additional targets to allow us to link against the
+libraries:
+1. `//tensorflow:tensorflow_dll_import_lib`
+2. `//tensorflow/compiler/tf2xla/xla_tensor:x10_dll_import_lib `
+
+We must pass the `--nocheck_visibility` flag to bazel to accomodate the new
+libraries.
+
+<details>
```

Comment:
Would it make sense to check in these build commands (for Linux/macOS/Windows) as an actual script file, instead of a README code snippet? That sounds more maintainable and testable.

Reply:
I thought about that - but I don't want people to blindly run the scripts, because Im not 100% certain on how generalized they are.  I think that at some point, it would make sense to extract them into scripts, but not right now.

Reply:
Okay! I think scripts may eventually be nice because people will almost certainly just copy-paste the entire block of commands from the documentation (extract the script). Otoh, keeping the commands in a documentation file provides more context for the script.